### PR TITLE
Example for entailment: SciTail

### PR DESCRIPTION
**Pull request by nomisto**
This is an example for an Entailment dataset: SciTail, see #30.

SciTail has four different source formats, so I am not sure which to use. Currently the `source` schema is from based on the TSV format (see below), which matches the format of the bigbio TE schema 1:1 (config `scitail_source` and config `scitail_bigbio_te` are the same). Is this alright or should I merge all of the formats below to one `source` format?

All formats:

TSV: premise, hypothesis, label,  **currently also used for `source`**
SNLI: sentence1_binary_parse, sentence1_parse, gold_label, sentence2_binary_parse, sentence1, annotator_labels, sentence2
dgem: premise, hypothesis, label, hypothesis graph structure
predictor: gold_label, sentence1, sentence2, sentence2_structure 

Example:

```batch
dgem

"premise":  An introduction to atoms and elements, compounds, atomic structure and bonding, the molecule and chemical reactions.
"hypothesis": Replace another in a molecule happens to atoms during a substitution reaction.
"label": neutral
"hypothesis graph structure": another<>in<>a molecule$$$another<>in a molecule happens to<>atoms$$$another<>in a molecule happens to atoms during<>a substitution reaction$$$a molecule<>happens to<>atoms$$$a molecule<>happens to atoms during<>a substitution reaction$$$atoms<>during<>a substitution reaction


predictor

"question":"What happens to atoms during a substitution reaction?",
"gold_label":"neutral",
"sentence1":"An introduction to atoms and elements, compounds, atomic structure and bonding, the molecule and chemical reactions.",
"answer":"replace another in a molecule",
"sentence2":"Replace another in a molecule happens to atoms during a substitution reaction.",
"sentence2_structure":"another<>in<>a molecule$$$another<>in a molecule happens to<>atoms$$$another<>in a molecule happens to atoms during<>a substitution reaction$$$a molecule<>happens to<>atoms$$$a molecule<>happens to atoms during<>a substitution reaction$$$atoms<>during<>a substitution reaction"}


snli

"sentence1_binary_parse":"( ( ( An introduction ) ( to ( ( ( ( ( ( ( ( ( ( atoms and ) elements ) , ) compounds ) , ) ( atomic ( ( structure and ) bonding ) ) ) , ) ( the molecule ) ) and ) ( chemical reactions ) ) ) ) . )",
"sentence1_parse":"(ROOT (NP (NP (DT An) (NN introduction)) (PP (TO to) (NP (NP (NNS atoms) (CC and) (NNS elements)) (, ,) (NP (NNS compounds)) (, ,) (NP (JJ atomic) (NN structure) (CC and) (NN bonding)) (, ,) (NP (DT the) (NN molecule)) (CC and) (NP (NN chemical) (NNS reactions)))) (. .)))",
"gold_label":"neutral",
"sentence2_binary_parse":"( Replace ( ( another ( in ( a molecule ) ) ) ( ( ( happens ( to atoms ) ) ( during ( a ( substitution reaction ) ) ) ) . ) ) )",
"sentence1":"An introduction to atoms and elements, compounds, atomic structure and bonding, the molecule and chemical reactions.",
"annotator_labels":["-","neutral","neutral","neutral","neutral"],
"sentence2_parse":"(ROOT (S (ADVP (RB Replace)) (NP (NP (DT another)) (PP (IN in) (NP (DT a) (NN molecule)))) (VP (VBZ happens) (PP (TO to) (NP (NNS atoms))) (PP (IN during) (NP (DT a) (NN substitution) (NN reaction)))) (. .)))",
"sentence2":"Replace another in a molecule happens to atoms during a substitution reaction."}


tsv

"premise":  An introduction to atoms and elements, compounds, atomic structure and bonding, the molecule and chemical reactions.
"hypothesis": Replace another in a molecule happens to atoms during a substitution reaction.
"label": neutral
```

### CODE REVIEW

**Comment by galtay** [comment]
sorry for all the back and forth in slack! but the new home for dataset loader scripts is back in the `examples` directory

**Comment by galtay** [comment]
you *should* be able to do something like 
```
python -m tests.test_bigbio --path examples/scitail.py  --view bigbio_entailment
```

**Comment by nomisto** [comment]
No worries! Tests are OK on both `bigbio_entailment` and `source`

File: examples/scitail.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+_HOMEPAGE = "https://allenai.org/data/scitail"
+
+_LICENSE = "Apache License 2.0"
+
+_URLS = {
+    _DATASETNAME: "https://ai2-public-datasets.s3.amazonaws.com/scitail/SciTailV1.1.zip",
+}
+
+_SUPPORTED_TASKS = ["entailment"]
```
Comment (galtay) [inline examples/scitail.py:61]: fresh of the presses, new and updated task names :) 
```suggestion
_SUPPORTED_TASKS = ["TE"]
```

File: examples/scitail.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="scitail_source",
+            version=SOURCE_VERSION,
+            description="SciTail source schema",
+            schema="source",
+            subset_id="scitail",
+        ),
+        BigBioConfig(
+            name="scitail_bigbio_entailment",
```
Comment (galtay) [inline examples/scitail.py:94]:
```suggestion
            name="scitail_bigbio_te",
```

File: examples/scitail.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            version=BIGBIO_VERSION,
+            description="SciTail BigBio schema",
+            schema="bigbio_entailment",
```
Comment (galtay) [inline examples/scitail.py:97]:
```suggestion
            schema="bigbio_te",
```

File: examples/scitail.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "label": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == "bigbio_entailment":
```
Comment (galtay) [inline examples/scitail.py:116]:
```suggestion
        elif self.config.schema == "bigbio_te":
```

File: examples/scitail.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        # since examples can contain quotes mid text set quoting to QUOTE_NONE (3) when reading tsv
+        # e.g.: ... and apply specific "tools" to examples and ... 
+        data = pd.read_csv(filepath, sep="\t", names=["premise", "hypothesis", "label"], quoting=3)
+        data["id"] = data.index
+
+        if self.config.schema == "source":
+            for _, row in data.iterrows():
+                yield row["id"], row.to_dict()
+
+        elif self.config.schema == "bigbio_entailment":
```
Comment (galtay) [inline examples/scitail.py:170]:
```suggestion
        elif self.config.schema == "bigbio_te":
```

**Comment by galtay** [comment]
when you get a chance can you please try, 
```
python -m tests.test_bigbio --path examples/scitail.py  --schema TE
```

**Comment by nomisto** [comment]
Test is passed

**Review by galtay — APPROVED** [review]
:tada:

**Pull request merged**